src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.cs

```diff
@@ -0,0 +1,248 @@
+                        context.ReportDiagnostic(operation.CreateDiagnostic(Rule));
+                        return;
+                    }
+                }
+            }
+        }
+
+        internal sealed class RequiredSymbols
+        {
+            //  Named-constructor 'TryGetSymbols' inits all properties or doesn't construct an instance.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+            private RequiredSymbols() { }
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+
+            public static bool TryGetSymbols(Compilation compilation, [NotNullWhen(true)] out RequiredSymbols? symbols)
+            {
+                symbols = default;
+
+                var stringType = compilation.GetSpecialType(SpecialType.System_String);
+                var boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+
+                if (stringType is null || boolType is null)
+                    return false;
+
+                if (!compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison, out var stringComparisonType))
+                    return false;
+
+                var compareMethods = stringType.GetMembers(nameof(string.Compare))
+                    .OfType<IMethodSymbol>()
+                    .Where(x => x.IsStatic);
+                var compareStringString = compareMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType);
+                var compareStringStringBool = compareMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType, boolType);
+                var compareStringStringStringComparison = compareMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType, stringComparisonType);
+
+                var equalsMethods = stringType.GetMembers(nameof(string.Equals))
+                    .OfType<IMethodSymbol>()
+                    .Where(x => x.IsStatic);
+                var equalsStringString = equalsMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType);
+                var equalsStringStringStringComparison = equalsMethods.GetFirstOrDefaultMemberWithParameterTypes(stringType, stringType, stringComparisonType);
+
+                symbols = new RequiredSymbols
+                {
+                    StringType = stringType,
+                    BoolType = boolType,
+                    StringComparisonType = stringComparisonType,
+                    CompareStringString = compareStringString,
+                    CompareStringStringBool = compareStringStringBool,
+                    CompareStringStringStringComparison = compareStringStringStringComparison,
+                    EqualsStringString = equalsStringString,
+                    EqualsStringStringStringComparison = equalsStringStringStringComparison
```

Comment:
We should check symbol requirements here for pairings. `CompareStringString` requires `EqualsStringString` for example. Return false if no valid configuration is available

Reply:
Good idea. I'll add that.